cells/_automation/nixago.nix

```diff
@@ -0,0 +1,55 @@
+{
+  inputs,
+  cell,
+}: let
+  inherit (inputs) nixpkgs;
+  inherit (inputs.std) std presets;
+in {
+  treefmt = presets.nixago.treefmt {
+    configData.formatter = {
+      prettier = {
+        # TODO: remove excludes
+        excludes = ["**.md" "**.toml"];
+      };
+      haskell = {
+        command = "ormolu";
+        options = [
+          "--ghc-opt"
+          "-XBangPatterns"
+          "--ghc-opt"
+          "-XPatternSynonyms"
+          "--ghc-opt"
+          "-XTypeApplications"
+          "--mode"
+          "inplace"
+          "--check-idempotence"
+        ];
+        # TODO: activate haskell formatter:
+        #includes = ["*.hs"];
+      };
+    };
+    packages = [nixpkgs.ormolu];
+  };
+
+  editorconfig = presets.nixago.editorconfig {
+    configData = {
```

Comment:
Isn't this duplicating the checked in config file?

Reply:
the checked-in config file is generated by this upon entering devshell.

Reply:
I think I still really don't get the point of `nixago`. Why are we checking it in if it's generated? or if it's checked in, why do we need to generate it?

Reply:
> Why are we checking it in if it's generated?

Not sure (cc/ @blaggacao) but I believe for online editors, and maybe also for users that did not use `direnv`/`nix develop` before opening their editor.

Reply:
I didn't see this comment before. Sry.

Sometimes you may enter an editor without previously having entered the devshell.

I thought it was still good to have editorconfig in place so that the basic formatting is respected.

Now that I think about it I may have simplified and assumed that this file is loaded upon _starting_ the editor. But now I realize that it doesn't make sense. It must be loaded every time opening a file.

So actually my initial reasoning here was flawed.

You can change this relatively easily with setting `hook.mode = "link";`. Might be worth a try. You'd also have to remove the file from git tracking manually first, since link mode registers them in gitignore.